### DISCUSSION TITLE
Remove FM from folder

### DISCUSSION THREAD

**Discussion by LionUCS** [estruyf/vscode-front-matter#826] [Q&A] [answered]
I initialized FM in the wrong folder. I deleted the .frontmatter directory and frontmatter.json, thinking that world remove it, but keeps coming back. How do I remove this?

**Comment by estruyf** [comment]
That should do the trick. Once you removed both of those, try restarting the editor. If that would still make it come back, check the `.vscode/settings.json` file if a `frontMatter` setting is accidentally configured. If that is the case, make sure to remove it as well.

**Comment by LionUCS** [comment]
forgot to mention I'm on windows. Anyway, I just tried removing some blank FM entries from settings, deleting the workspace, even deleting all the workspaces -- the .frontmatter directory still comes back.

If there is a way I would like to know, but its not that important -- its just a tiny bit annoying and I'll be installing linux on this system eventually

**Reply by estruyf** [reply]
Still, I wonder why it automatically comes back. When you open the Front Matter output, what does it show you (https://frontmatter.codes/docs/troubleshooting#logging)?

**Comment by LionUCS** [comment, marked as answer]
According to the logs, it appears to be finding frontmatter in a subfolder

```
["INFO" - 10:41:16] VSCODE | Activating Front Matter CMS version 10.2.0...
["INFO" - 10:41:16] VSCODE | Logging level: info
["INFO" - 10:41:17] VSCODE | No ".frontmatter/config" config files found.
["INFO" - 10:41:17] VSCODE | Project config path: c:\████\██████\██████████████\frontmatter.json
["INFO" - 10:41:18] VSCODE | Triggering listener: settings-init
["INFO" - 10:41:18] VSCODE | Triggering listener: settings-global
["INFO" - 10:41:18] VSCODE | Sending message to panel: folderInfo
["INFO" - 10:41:18] VSCODE | Triggering listener: settings-init
["INFO" - 10:41:18] VSCODE | Triggering listener: settings-global
["INFO" - 10:41:18] VSCODE | Sending message to panel: folderInfo
```

If thats the case, that's less annoying. Though I was hoped the multi folder feature would work -- I'm assuming its still in beta

**Reply by LionUCS** [reply]
So I tried opening another folder where I know is ancestor to a project with FM
Similar result, however it seems to want the registered folders to start from the workspace

```
["INFO" - 11:08:21] VSCODE | Activating Front Matter CMS version 10.2.0...
["INFO" - 11:08:21] VSCODE | Logging level: info
["INFO" - 11:08:21] VSCODE | No ".frontmatter/config" config files found.
["INFO" - 11:08:21] VSCODE | Project config path: c:\████\██████\███████\████\frontmatter.json
["ERROR" - 11:08:21] VSCODE | Front Matter CMS: Folder "wiki ([[workspace]]/src/wiki)" does not exist. Please remove it from the settings.
["ERROR" - 11:08:21] VSCODE | Front Matter CMS: Folder "blog ([[workspace]]/src/blog)" does not exist. Please remove it from the settings.
["INFO" - 11:08:22] VSCODE | Triggering listener: settings-init
["INFO" - 11:08:22] VSCODE | Triggering listener: settings-global
["INFO" - 11:08:22] VSCODE | Sending message to panel: folderInfo
["INFO" - 11:08:22] VSCODE | Triggering listener: settings-init
["INFO" - 11:08:22] VSCODE | Triggering listener: settings-global
["INFO" - 11:08:22] VSCODE | Sending message to panel: folderInfo
```

Would it not be better to use the `frontmatter.json` as the anchor?

**Reply by estruyf** [reply]
If you use workspaces in VS Code, the `fronmatter.json` file is an anchor; otherwise, the project path will be used.

Now I also understand why the `frontmatter.json` file gets created; that is indeed a bug that needs to be solved.

**Reply by estruyf** [reply]
Includes a fix in the latest beta to not recreate the `frontmatter.json` file when one exist in a sub-folder.
